libconsensus/pbft/PBFTSealer.h

```diff
@@ -89,32 +75,56 @@ class PBFTSealer : public Sealer
         return m_sealing.block.blockHeader().number() == (m_blockChain->number() + 1);
     }
 
-    bool reachBlockIntervalTime() override { return m_pbftEngine->reachBlockIntervalTime(); }
-
-private:
-    bool inline isLeaderOrNextLeader()
+    bool reachBlockIntervalTime() override
     {
-        return (m_pbftEngine->getLeader().second == m_pbftEngine->nodeIdx() ||
-                m_pbftEngine->getNextLeader() == m_pbftEngine->nodeIdx());
+        return m_pbftEngine->reachBlockIntervalTime() || canFinishSealing();
     }
-    /*
-    bool emptyTxPool()
+    /// in case of the next leader packeted the number of maxTransNum transactions before the last
+    /// block is consensused
+    bool canHandleBlockForNextLeader() override
+    {
+        return m_pbftEngine->canHandleBlockForNextLeader();
+    }
+
+private:
+    /**
+     * @brief : decide can finish sealing or not
+     *          1. the block is not an empty block(namely the txqueue size is equal to the
+     * transaction size of the block)
+     *          2. the transaction pool is empty
+     * @return true : can  finish sealing
+     * @return false : can't finish sealing
+     */
+    bool inline canFinishSealing()
     {
         return (m_sealing.block.getTransactionSize() > 0) &&
-               m_txPool->status().current == m_sealing.m_transactionSet.size();
-    }*/
-    void setBlock();
+               (m_txPool->status().current == m_sealing.block.getTransactionSize());

```

Comment:
No need to call getTxSize in one sentence

And how about to finish sealing whenever the getTxSize > 0?